### DISCUSSION TITLE
Argument of type 'StateCreator<test, [], [["zustand/persist", unknown]]>' is not assignable to parameter of type 'StateCreator<test, [], []>'.

### DISCUSSION THREAD

**Discussion by leonsama** [pmndrs/zustand#2922] [Bug report]
### Bug Description

Following the example in [https://zustand.docs.pmnd.rs/integrations/persisting-store-data](https://zustand.docs.pmnd.rs/integrations/persisting-store-data)

<img width="531" alt="{196C03BF-CCA9-420F-8609-3B1BA0A0BB6B}" src="https://github.com/user-attachments/assets/faf6e020-2750-4755-920d-f3dba5555676" />

### Reproduction Link

https://stackblitz.com/edit/vitejs-vite-jsze76eg?file=src%2FApp.tsx&terminal=dev

**Comment by dbritto-dev** [comment]
@SolomonLeon on TypeScript you should use `create()(...)` instead of `create(...)` the code example was written for JavaScript, apart from that there's a typo in your demo `test` should be written in pascalcase and is `addABear` not `addABears`

**Reply by leonsama** [reply]
Sorry for my typo, I've made the changes and it works on [stackblitz](https://stackblitz.com/edit/vitejs-vite-jsze76eg?file=src%2FApp.tsx&terminal=dev), however when I clome the same code to my local vscode editor, it reports the same error:

```
Argument of type 'StateCreator<test, [], [["zustand/persist", unknown]]>' is not assignable to parameter of type 'StateCreator<test, [], [never, unknown][]>'.
  Type 'StateCreator<test, [], [["zustand/persist", unknown]]>' is not assignable to type '{ $$storeMutators?: [never, unknown][] | undefined; }'.
    Types of property '$$storeMutators' are incompatible.
      Type '[["zustand/persist", unknown]] | undefined' is not assignable to type '[never, unknown][] | undefined'.
        Type '[["zustand/persist", unknown]]' is not assignable to type '[never, unknown][]'.
          Type '["zustand/persist", unknown]' is not assignable to type '[never, unknown]'.
            Type at position 0 in source is not compatible with type at position 0 in target.
              Type '"zustand/persist"' is not assignable to type 'never'.ts(2345)
```

Do you have any ideas about that?

**Reply by dbritto-dev** [reply]
@SolomonLeon try restarting ts server
